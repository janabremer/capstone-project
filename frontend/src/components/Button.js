import styled from 'styled-components/macro'
import PropTypes from 'prop-types'

Button.propTypes = {
    onClick: PropTypes.func,
    text: PropTypes.string.isRequired,
}

export default function Button({onClick, text, iconComponentLeft, iconComponentRight}) {
    return(
        <ButtonStyled onClick={onClick}>
            {iconComponentLeft}
            <span>{text}</span>
            {iconComponentRight}
        </ButtonStyled>
    )
}

const ButtonStyled = styled.button`
    font-size: inherit;
    color: var(--primary-blue);
    padding: 10px 20px;
    border: none;
    border-radius: 10px;
    background-color: var(--primary-light-transparent);
    display: flex;
    place-items: center;
`
