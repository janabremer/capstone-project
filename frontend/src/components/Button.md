```js
<Button handleClick={} text={'Click me'} />
```
