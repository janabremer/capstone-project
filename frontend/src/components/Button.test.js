import { render } from '@testing-library/react'
import user from '@testing-library/user-event'
import 'jest-styled-components'
import Button from './Button'

const testProps = {
    text: 'Click me',
    onClick: () => {}
}

describe('Button', () => {
    it('renders correctly', () => {
        const { container } = render(
            <Button {...testProps}/>
        )
        expect(container.firstChild).toMatchSnapshot()
    })

    it('calls onClickMock on click', () => {
        const onClickMock = jest.fn()
        const props = {...testProps, onClick: onClickMock}
        const { getByRole } = render(<Button {...props}/>)
        const button = getByRole('button')
        user.click(button)
        expect(onClickMock).toHaveBeenCalledTimes(1)
    })
})