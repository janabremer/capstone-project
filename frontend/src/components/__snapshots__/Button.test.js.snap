// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  font-size: inherit;
  color: var(--primary-blue);
  padding: 10px 20px;
  border: none;
  border-radius: 10px;
  background-color: var(--primary-light-transparent);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  place-items: center;
}

<button
  class="c0"
>
  <span />
</button>
`;
